client/views/admin/settings/groups/LDAPGroupPage.tsx

```diff
@@ -45,28 +49,70 @@ function LDAPGroupPage({ _id, ...group }: ISetting): JSX.Element {
 	const handleSyncNowButtonClick = async (): Promise<void> => {
 		try {
 			await testConnection(undefined);
-			// #ToDo: Switch to modal.setModal
-			modal.open(
-				{
-					title: t('Execute_Synchronization_Now'),
-					text: t('LDAP_Sync_Now_Description'),
-					confirmButtonText: t('Sync'),
-					showCancelButton: true,
-					closeOnConfirm: true,
-					closeOnCancel: true,
-				},
-				async (isConfirm: boolean): Promise<void> => {
-					if (!isConfirm) {
-						return;
-					}
+			const confirmSync = async (): Promise<void> => {
+				try {
+					const { message } = await syncNow(undefined);
+					dispatchToastMessage({ type: 'success', message: t(message) });
+				} catch (error) {
+					dispatchToastMessage({ type: 'error', message: error });
+				}
+			};
 
-					try {
-						const { message } = await syncNow(undefined);
-						dispatchToastMessage({ type: 'success', message: t(message) });
-					} catch (error) {
-						dispatchToastMessage({ type: 'error', message: error });
-					}
-				},
+			setModal(
+				<GenericModal
+					variant='info'
+					confirmText={t('Sync')}
+					cancelText={t('Cancel')}
+					title={t('Execute_Synchronization_Now')}
+					onConfirm={confirmSync}
+					onClose={closeModal}
+				>
+					<Box>{t('LDAP_Sync_Now_Description')}</Box>
+				</GenericModal>,
+			);
+		} catch (error) {
+			dispatchToastMessage({ type: 'error', message: error });
+		}
+	};
+
+	const handleSearchTestButtonClick = async (): Promise<void> => {
+		try {
+			await testConnection(undefined);
+			let username = '';
+			const handleChangeUsername = (event: FormEvent<HTMLInputElement>): void => {
+				username = event.currentTarget.value;
+			};
+
+			const confirmSearch = async (): Promise<void> => {
+				try {
+					const { message } = await testSearch({ username });
+					dispatchToastMessage({ type: 'success', message: t(message) });
+				} catch (error) {
+					dispatchToastMessage({ type: 'error', message: error });
+				}
+			};
+
+			setModal(
+				<GenericModal
+					variant='info'
+					confirmText={t('Search')}
+					cancelText={t('Cancel')}
+					title={t('Test_LDAP_Search')}
+					onConfirm={confirmSearch}
+					onClose={closeModal}
+				>
+					<Field>
+						<Flex.Container>
```

Comment:
I'm sorry, we hadn't communicated it in the code yet: `Flex.Container` and `Flex.Item` are deprecated. I suggest using `display='flex'` on the child `Box`.